opencensus/common/monitored_resource_util/monitored_resource.py

```diff
@@ -12,75 +12,76 @@
 _AWS_EC2_INSTANCE = "aws_ec2_instance"
 
+# Kubenertes environment variables
+_KUBERNETES_SERVICE_HOST = 'KUBERNETES_SERVICE_HOST'
+
 
-class MonitoredResource(object):
-    """MonitoredResource returns the resource type and resource labels.
+def is_gke_environment():
+    """A Google Container Engine (GKE) container instance.
+    KUBERNETES_SERVICE_HOST environment variable must be set.
     """
+    return _KUBERNETES_SERVICE_HOST in os.environ
 
-    @property
-    def resource_type(self):
-        """Returns the resource type this MonitoredResource.
-        :return:
-        """
-        raise NotImplementedError  # pragma: NO COVER
 
-    def get_resource_labels(self):
-        """Returns the resource labels for this MonitoredResource.
-        :return:
-        """
-        raise NotImplementedError  # pragma: NO COVER
+def is_gce_environment():
+    """A virtual machine instance hosted in Google Compute Engine (GCE)."""
+    return gcp_metadata_config.GcpMetadataConfig.is_running_on_gcp()
 
 
-class GcpGceMonitoredResource(MonitoredResource):
-    """GceMonitoredResource represents gce_instance type monitored resource.
-    For definition refer to
-    https://cloud.google.com/monitoring/api/resources#tag_gce_instance
-    """
+def is_aws_environment():
+    """A virtual machine instance in Amazon EC2"""
+    return aws_identity_doc_utils.AwsIdentityDocumentUtils.is_running_on_aws()
 
-    @property
-    def resource_type(self):
-        return _GCE_INSTANCE
 
-    def get_resource_labels(self):
-        gcp_config = GcpMetadataConfig()
-        return gcp_config.get_gce_metadata()
+def get_instance():
+    """Get a monitored resource based on the application environment.
 
+    Returns a self-configured monitored resource, or None if the application is
+    not running on a supported environment.
 
-class GcpGkeMonitoredResource(MonitoredResource):
-    """GkeMonitoredResource represents gke_container type monitored resource.
-    For definition refer to
-    https://cloud.google.com/monitoring/api/resources#tag_gke_container
-    """
-
-    @property
-    def resource_type(self):
-        return _GKE_CONTAINER
+    It supports following environments (resource types)
+    1. gke_container:
+    2. gce_instance:
+    3. aws_ec2_instance:
 
-    def get_resource_labels(self):
-        gcp_config = GcpMetadataConfig()
-        return gcp_config.get_gke_metadata()
+    AwsMonitoredResource represents aws_ec2_instance type monitored resource.
+    For definition refer to
+    https://cloud.google.com/monitoring/api/resources#tag_aws_ec2_instance
 
+    GkeMonitoredResource represents gke_container type monitored resource.
+    For definition refer to
+    https://cloud.google.com/monitoring/api/resources#tag_gke_container
 
-class AwsMonitoredResource(MonitoredResource):
-    """AwsMonitoredResource represents aws_ec2_instance type monitored resource.
+    GceMonitoredResource represents gce_instance type monitored resource.
```

Comment:
TODO: update the docs to remove references to `MonitoredResource` classes and fix return type here.